Modules/IO/NIFTI/test/itkNiftiReadWriteDirectionTest.cxx

```diff
@@ -205,17 +205,45 @@ itkNiftiReadWriteDirectionTest(int argc, char * argv[])
   if (!itk::ExposeMetaData<std::string>(dictionary, "ITK_sform_corrected", temp) || temp != "YES")
   {
     std::cerr << "ITK_sform_corrected metadata flag was not properly set" << std::endl;
-    std::cerr << " expected YES, recieved:" << temp.c_str() << std::endl;
+    std::cerr << " expected YES, received:" << temp.c_str() << std::endl;
     return EXIT_FAILURE;
   }
 
+  itksys::SystemTools::PutEnv("ITK_NIFTI_SFORM_PERMISSIVE=NO");
+
   // check the resulting rotation matrix is orthogonal
   if (!CheckRotation<TestImageType>(inputImageNonOrthoSform))
   {
     std::cerr << "Rotation matrix after correcting is not orthgonal" << std::endl;
     return EXIT_FAILURE;
   }
 
+  // check similarity between direction without shear (argv[2]) and direction with shear (argv[4])
+  // inputImageNoQformDirection should be somewhat similar to inputImageNonOrthoSformDirection
+  // the shear in the test data is pretty large, so they are off by a bit, but check angle to catch rotation flips
+  // introduced by previous bugs
+  auto inputImageNonOrthoSformDirection = inputImageNonOrthoSform->GetDirection();
+
+  // compare the direction matrices
+  auto nonOrthoDirectionInv = inputImageNonOrthoSformDirection.GetTranspose();
+
+  // Compute R_relative = R1 * R2^-1
+  auto rRelative = inputImageNoQformDirection * nonOrthoDirectionInv;
+
+  // Calculate the trace of the relative rotation matrix
+  double trace = rRelative[0][0] + rRelative[1][1] + rRelative[2][2];
+
+  // Calculate the angle of rotation between the two matrices
+  double angle = std::acos((trace - 1.0) / 2.0);
+
+  if (angle > 0.15)
```

Comment:
Is this particular value chosen for some reason? If it is an arbitrary value, even if not used elsewhere, it should probably be defined as a variable and a comment be added to make it clear that it is a sensitive value. Better than that it would be to define a variable in the corresponding class, as commented in the issue.

Also, please print the value of the threshold in the error message.

Reply:
It's an arbitrary value, somewhat larger than the actual error from permissive mode (~0.1) but small enough to catch the bug that is fixed in this PR. The threshold will depend on the input data. I agree that if we switch from permissive mode on/off to a numeric threshold, it would make sense to se that in the tests. Meanwhile, I'll make it a local variable with a comment

Reply:
Alternatively I could make it an argument to the script, if that is OK? `itkNiftiReadWriteDirectionTest` is called a couple of times with different images, it might make sense to use different thresholds

Reply:
Maybe an optional one? If we eventually transition towards a CMake variable or a class ivar, the test can be modified to adopt that one if not provided as an input argument?

Reply:
I made the threshold an argument to the script, with a default.

Reply:
Thanks @cookpa. Out of curiosity, why are the values one magnitude larger than the threshold that you used previously (re https://github.com/InsightSoftwareConsortium/ITK/pull/4847#discussion_r1759588013)?

Reply:
I changed the threshold to degrees when making it a command line parameter, I figured it would be easier

Reply:
:+1: Oh, should have figured out. Thanks Philip.

Reply:
No problem, thanks for your suggestions